### PR TITLE
Added mock method and static response for POST to /token

### PR DESCRIPTION
**Pull request by goshe**
Cereated a postToken command in view/ (would be a different directory like action / post better ?)
- Accepts client_id, username, password to verify the user
- Returns 403 if parameters are missing or password does not match
- Returns a 200 and jwt on success (ClientId, User and Password match)
- Supports only the grant type password for now

Additional Changes:
- Enhanced user mock by adding a password field.
- Created PostFn as Type to support a body field.
- Added tests for 403 cases and 200.

Please review and add comments where I did not understand the structure well, or where I should change methods / names / folders.

Cheers,
Rene

### CODE REVIEW

**Comment by goshe** [comment]
No merge needed, yet, I would like to finish the implementation with a proper return value for access and request token. 
But I am happy to take your feedback :-)

File: lib/types.ts [hunk 1]
```diff
@@ -18,3 +18,9 @@ export type MiddlewareFn = (
   instance: MockInstance,
   request: NockClientRequest
 ) => Promise<void>;
+
+export type PostFn = (
```
Comment (Photonios) [inline lib/types.ts]: Can you rename this to `PostViewFn` ?

File: lib/views/postToken.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { PostFn } from "../types";
+import createBearerToken, {CreateTokenOptions} from "../createBearerToken"
+
+const postToken: PostFn = (instance, request, requestBody) => {
+  if (typeof requestBody !== "object") {
+    return [403, "Access denied"];
```
Comment (Photonios) [inline lib/views/postToken.ts]: Shouldn't the status code be 400 ?

File: lib/views/postToken.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  const { username, password, client_id } = (requestBody as unknown) as Record<
+    string,
+    any
+  >;
+  if (!username || !password || instance.params.clientID !== client_id) {
+    return [403, "Access denied"];
+  }
+
+  let user = instance.database.findUserByEmail(username);
```
Comment (Photonios) [inline lib/views/postToken.ts]: can be `const`, same for `access_token` and `refresh_token`

File: lib/views/postToken.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  if (!user || password !== user.password) {
+    return [403, "Access denied"];
+  }
+
+  let access_token = instance.createBearerToken(user.id);
```
Comment (Photonios) [inline lib/views/postToken.ts]: The expiry time that is included in the response must also be part of the token. `createBearerToken` takes a second argument with the expiry time, default is 3600.

With the current implementation, there's a mismatch. The response says the token is valid for 300 seconds, while the token is actually valid for 3600 seconds. Same applies to the refresh token.

**Comment by Photonios** [comment]
I have merged these changes manually into `master` and patched them up. They are available in version 1.0.1 on NPM: https://www.npmjs.com/package/keycloak-mock/v/1.0.1

Thanks for the contribution 👍

**Comment by goshe** [comment]
Heya,

Cool thank you :-)
I am a bit busy with normal business but will need more stuff in the mock, so it wont be my last :-)

**Pull request closed without merging**